### PR TITLE
Create Setup Graft Supernode (Building from Source).md

### PR DESCRIPTION
**Pull request by mv1879**

### CODE REVIEW

**Review by SomethingGettingWrong — COMMENTED** [review]
Might be good to suggest they shouldn't run from home. That it takes numerous amounts of bandwidth but if they can make sure to forward the appropriate ports.

**Comment by yidakee** [comment]
I ported the "Alpha3 Supernode Setup for Dummies Guide" here -> https://github.com/graft-community/docs/pull/1

**Comment by jagerman** [comment]
This is a nice concise version for people who have already done it once or twice before (or are otherwise familiar with linux commands).  I think there is place for both this and the Dummies guide.

Perhaps for a v2 of each we can incorporate bits the other one is missing.

File: Setup Graft Supernode (Building from Source).md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+free -h
+df -h
+sudo fallocate -l 4G /swapfile          # Change 4G to the proper size swap file for your server
+sudo chmod 600 /swapfile
+sudo mkswap /swapfile
+sudo swapon /swapfile
+sudo su -
+cat <<EOF >> /etc/fstab
+/swapfile none swap sw 0 0
+EOF
```
Comment (jagerman) [inline Setup Graft Supernode (Building from Source).md]: You can avoid the `sudo su -` here with this trick:

`echo '/swapfile none swap sw 0 0' | sudo tee -a /etc/fstab`

File: Setup Graft Supernode (Building from Source).md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+## Add Firewall
+
+
+````bash
+sudo ufw default allow outgoing
+sudo ufw default deny incoming
+sudo ufw allow ssh/tcp
+sudo ufw limit ssh/tcp
+sudo ufw allow 28600/tcp            # Load balancer http port
+sudo ufw allow 28643/tcp            # Load balancer https port
```
Comment (jagerman) [inline Setup Graft Supernode (Building from Source).md]: Are these two ports used for anything?

File: Setup Graft Supernode (Building from Source).md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+sudo ufw allow 28680/tcp            # p2p node port
+sudo ufw allow 28680/udp            # p2p node port
```
Comment (jagerman) [inline Setup Graft Supernode (Building from Source).md]: This one isn't needed (graftnoded only listens on tcp)

File: Setup Graft Supernode (Building from Source).md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+sudo ufw allow 28681/tcp            # rpc node port
+sudo ufw allow 28690/tcp            # rpc supernode port
+sudo ufw allow 8081/tcp             # block explorer port
```
Comment (jagerman) [inline Setup Graft Supernode (Building from Source).md]: Also unneeded.  (If you really want to run your own block explorer you should probably already know what you're doing).

File: Setup Graft Supernode (Building from Source).md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+sudo sed -i -e 's/.*DefaultLimitNOFILE=.*/DefaultLimitNOFILE=8192/g' /etc/systemd/system.conf
+````
+
+
+## Git Graft
+
+````bash
+git clone --recursive -b alpha3 https://github.com/graft-project/graft-ng.git
+cd graft-ng
+git submodule update --init --recursive
```
Comment (jagerman) [inline Setup Graft Supernode (Building from Source).md]: The submodule update isn't needed with `--recursive` above

File: Setup Graft Supernode (Building from Source).md (hunk 1 above)
Comment (mv1879) [reply Setup Graft Supernode (Building from Source).md]: Updated.

File: Setup Graft Supernode (Building from Source).md (hunk 3 above)
Comment (mv1879) [reply Setup Graft Supernode (Building from Source).md]: Updated.

File: Setup Graft Supernode (Building from Source).md (hunk 4 above)
Comment (mv1879) [reply Setup Graft Supernode (Building from Source).md]: Updated.

File: Setup Graft Supernode (Building from Source).md (hunk 5 above)
Comment (mv1879) [reply Setup Graft Supernode (Building from Source).md]: Removed submodule update.

**Pull request merged**